x/concentrated-liquidity/lp_test.go

```diff
@@ -276,11 +280,11 @@ func (s *KeeperTestSuite) TestCreatePosition() {
 			s.Require().Equal(userBalancePrePositionCreation.Sub(sdk.NewCoins(sdk.NewCoin(ETH, asset0), (sdk.NewCoin(USDC, asset1)))).String(), userBalancePostPositionCreation.String())
 			s.Require().Equal(poolBalancePrePositionCreation.Add(sdk.NewCoin(ETH, asset0), (sdk.NewCoin(USDC, asset1))).String(), poolBalancePostPositionCreation.String())
 
-			hasPosition := clKeeper.HasFullPosition(s.Ctx, tc.poolId, s.TestAccs[0], tc.lowerTick, tc.upperTick, defaultJoinTime, tc.freezeDuration)
+			hasPosition := clKeeper.HasFullPosition(s.Ctx, tc.poolId, s.TestAccs[0], tc.lowerTick, tc.upperTick, defaultJoinTime, tc.freezeDuration, tc.positionId)
 			s.Require().True(hasPosition)
 
 			// Check position state
-			s.validatePositionUpdate(s.Ctx, tc.poolId, s.TestAccs[0], tc.lowerTick, tc.upperTick, defaultJoinTime, tc.freezeDuration, tc.liquidityAmount)
+			s.validatePositionsUpdate(s.Ctx, tc.poolId, s.TestAccs[0], tc.lowerTick, tc.upperTick, tc.liquidityAmount)
```

Comment:
Why can't we continue using `validatePositionUpdate`? It looks like this change has led to introducing a getter in production code

Reply:
the test case `liquidityAmount` was combining the position 1 liquidity and position 2 liquidity because we previously merged this into one single position. Because of this, we validate all positions with the same key (minus ID). The alternative would be to just validate the second position, but this feels wrong.

Also, what is wrong with have a getter in production code?

Reply:
I guess what I could do is just each position individually for the test case that has two positions, but just felt less clean

Reply:
I ended up just checking the second position created for that single test case https://github.com/osmosis-labs/osmosis/pull/4661/commits/c906adae1baa2cf19a090ec602b4e8e61fa6db9d